tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/coprocessor/janitor/HBaseTransactionPruningPlugin.java

```diff
@@ -179,15 +180,17 @@ public void pruneComplete(long time, long maxPrunedInvalid) throws IOException {
       return;
     }
 
-    // Get regions for given time, so as to not delete them
+    // Get regions for the given time, so as to not delete them. The prune upper bounds for regions are recorded
+    // by TransactionProcessor and the deletion is done by this class. To avoid update/delete race condition,
+    // we only delete stale regions.
```

Comment:
OK, the comment makes this clear that it is because of race condition. Not sure what kind of race conditions? Also, I believe my comment still holds that this means, as long as a region exists, we will never delete its entries here. Do we really have to keep a record of a region's existence for every single pruneInterval since the beginning of time? That can be a lot of records over time.